util/rtt_darwin.go

```
@@ -0,0 +1,43 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package util
+
+import (
+	"syscall"
+
+	"golang.org/x/sys/unix"
+)
+
+func getConnRTT(raw syscall.RawConn) (*RTTInfo, error) {
+	var info *unix.TCPConnectionInfo
+	var getSockoptErr error
+	err := raw.Control(func(fd uintptr) {
+		info, getSockoptErr = unix.GetsockoptTCPConnectionInfo(int(fd), unix.IPPROTO_TCP, unix.TCP_CONNECTION_INFO)
+	})
+	if err != nil {
+		return nil, err
+	}
+	if getSockoptErr != nil {
+		return nil, getSockoptErr
+	}
+	var ret RTTInfo
+	if info != nil {
```

Under what circumstances is `info` nil?  Should we return an error?